From: wangfushuai <wangfushuai@baidu.com>
Subject: fs/proc/task_mmu: add VM_SHADOW_STACK for arm64 when support GCS
Date: Sat, 7 Jun 2025 21:15:25 +0800

The recent commit adding VM_SHADOW_STACK for arm64 GCS did not update
the /proc/[pid]/smaps display logic to show the "ss" flag for GCS pages.
This patch adds the necessary condition to display "ss" flag.

Link: https://lkml.kernel.org/r/20250607131525.76746-1-wangfushuai@baidu.com
Fixes: ae80e1629aea ("mm: Define VM_SHADOW_STACK for arm64 when we support GCS")
Signed-off-by: wangfushuai <wangfushuai@baidu.com>
Cc: Andrii Nakryiko <andrii@kernel.org>
Cc: Christophe Leroy <christophe.leroy@csgroup.eu>
Cc: David Hildenbrand <david@redhat.com>
Cc: Liam Howlett <liam.howlett@oracle.com>
Cc: Oscar Salvador <osalvador@suse.de>
Cc: Mark Brown <broonie@kernel.org>
Cc: Catalin Marinas <catalin.marinas@arm.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 fs/proc/task_mmu.c |    3 +++
 1 file changed, 3 insertions(+)

--- a/fs/proc/task_mmu.c~fs-proc-task_mmu-add-vm_shadow_stack-for-arm64-when-support-gcs
+++ a/fs/proc/task_mmu.c
@@ -994,6 +994,9 @@ static void show_smap_vma_flags(struct s
 #ifdef CONFIG_ARCH_HAS_USER_SHADOW_STACK
 		[ilog2(VM_SHADOW_STACK)] = "ss",
 #endif
+#if defined(CONFIG_ARM64_GCS)
+		[ilog2(VM_SHADOW_STACK)] = "ss",
+#endif
 #if defined(CONFIG_64BIT) || defined(CONFIG_PPC32)
 		[ilog2(VM_DROPPABLE)] = "dp",
 #endif
_
